lib/MusicBrainz/Server/Data/ArtistCredit.pm

```diff
@@ -12,11 +12,17 @@ use MusicBrainz::Server::Data::Utils qw( object_to_ids placeholders load_subobje
 use MusicBrainz::Server::Constants qw( entities_with );
 
 extends 'MusicBrainz::Server::Data::Entity';
-with 'MusicBrainz::Server::Data::Role::EntityCache';
+with 'MusicBrainz::Server::Data::Role::EntityModelClass';
+with 'MusicBrainz::Server::Data::Role::GetByGID';
+with 'MusicBrainz::Server::Data::Role::MainTable';
+# Follows GetByGID and MainTable as it requires 'get_by_gid' and '_main_table';
```

Comment:
Have you tried just consuming them all in one `with` call instead? I did that for all of `Data::` in https://github.com/metabrainz/musicbrainz-server/pull/2751 and it didn't seem to cause issues (while avoiding the need to order things in weird and specific ways) :)

Reply:
No I just followed the current coding style for now even though it isn’t perfect. Changing it isn’t straightforward either. For example “the need of order things” is still there in the case when two roles like `GIDEntityCache` and `GIDRedirect` goes around the same method `get_by_gid`.